src/components/CardContainer.tsx

```diff
@@ -0,0 +1,202 @@
+  badge?: {
+    type: 'default' | 'error' | 'success' | 'warning' | 'info' | 'highlight';
+    label: string;
+    hasIcon?: boolean;
+  };
+  footer?:
+    | {
+        action1: {
+          onClick: () => void;
+          title: string;
+        };
+        text: string;
+      }
+    | {
+        action1: {
+          onClick: () => void;
+          title: string;
+        };
+        action2: {
+          onClick: () => void;
+          title: string;
+        };
+      };
+};
+
+type BadgeProps = {
+  icon: React.ElementType;
+  backgroundColor: string;
+  borderColor: string;
+  fontColor: string;
+};
+
+const getBadgeProps = (type: string, palette: Palette): BadgeProps => {
+  switch (type) {
+    case 'success':
+      return {
+        icon: CheckIcon,
+        backgroundColor: palette.base?.green[100]!,
+        borderColor: palette.base?.green[200]!,
+        fontColor: palette.base?.green[800]!,
+      };
+    case 'warning':
+      return {
+        icon: WarningIcon,
+        backgroundColor: palette.base?.yellow[100]!,
+        borderColor: palette.base?.yellow[200]!,
+        fontColor: palette.base?.yellow[800]!,
+      };
+    case 'info':
+      return {
+        icon: InfoIcon,
+        backgroundColor: palette.base?.lightBlue[100]!,
+        borderColor: palette.base?.lightBlue[200]!,
+        fontColor: palette.base?.lightBlue[800]!,
+      };
+    case 'highlight':
+      return {
+        icon: AddIcon,
+        backgroundColor: palette.base?.lilac[100]!,
+        borderColor: palette.base?.lilac[200]!,
+        fontColor: palette.base?.lilac[800]!,
+      };
+    case 'error':
+    default:
+      return {
+        icon: ErrorIcon,
+        backgroundColor: palette.base?.red[100]!,
+        borderColor: palette.base?.red[200]!,
+        fontColor: palette.base?.red[800]!,
+      };
+  }
+};
+
+const getFooterActions = (footer: CardContainerProps['footer']) => {
+  if (!footer) return null;
+  if ('action2' in footer) {
+    return (
+      <>
+        <Button
+          variant="text"
+          {...footer.action2}
+        >
+          {footer.action2.title}
+        </Button>
+        <Button
+          variant="contained"
+          {...footer.action1}
+        >
+          {footer.action1.title}
+        </Button>
+      </>
+    );
+  } else {
+    return (
+      <>
+        <Typography
+          variant="globalXXS"
+          color="textSecondary"
+        >
+          {footer.text}
+        </Typography>
+        <Button // update to new buttons
+          variant="text"
+          {...footer.action1}
+          endIcon={<ChevronRight fontSize="small" />}
+        >
+          {footer.action1.title}
+        </Button>
+      </>
+    );
+  }
+};
+
+const CardContainer = ({
+  badge = undefined,
+  footer = undefined,
+  children,
+  sx,
+  ...props
+}: CardContainerProps) => {
+  const theme = useTheme();
+  const badgeProps = badge ? getBadgeProps(badge.type, theme.palette) : null;
```

Comment:
:star: Acá haría el chequeo de que `badge` exista dentro de la función. Como está en el `getFooterActions`. Creo que no haría falta pasarle todo el `badge` para poder hacer ese chequeo, le pasas el `type` y chequeas que ese exista.

```suggestion
  const badgeProps = getBadgeProps(badge?.type, theme.palette);
```